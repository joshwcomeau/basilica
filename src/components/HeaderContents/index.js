// eslint-disable-next-line no-unused-vars
import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

import Badge from '../Badge';
import Icon from '../Icon';
import InnerWrapper from '../InnerWrapper';
import './index.scss';


const HeaderContents = ({ toggleCart, numOfCartItems }) => {
  const classes = classNames(['header-contents']);

  const badge = numOfCartItems > 0 && (
    <Badge className="cart-badge">{numOfCartItems}</Badge>
  );

  return (
    <div className={classes}>
      <InnerWrapper>
        <button className="cart-button" onClick={toggleCart}>
          {badge}
          <span className="cart-text">CART</span>
          <Icon value="shopping_cart" size={16} />
        </button>
      </InnerWrapper>
    </div>
  );
};

HeaderContents.propTypes = {
  numOfCartItems: PropTypes.number,
  toggleCart: PropTypes.func,
};

export default HeaderContents;
