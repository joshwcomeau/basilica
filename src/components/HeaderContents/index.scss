@import '../../variables';
@import '../../mixins';

.header-contents {
  position: fixed;
  z-index: 7;
  height: $header-height;
  line-height: $header-height;
  top: 0;
  left: 0;
  right: 0;
  text-align: right;
  padding: 0 $main-content-border-size * 2;
  max-width: $max-width;
  margin: 0 auto;

  .cart-button {
    position: relative;
    border: none;
    background: transparent;
    padding: 6px 5px;
    font-size: 16px;
    cursor: pointer;
    outline: none;

    .cart-text {
      display: inline-block;
      margin-right: 4px;
    }

    .icon {
      vertical-align: top;
    }
  }
  .cart-badge {
    position: absolute;
    top: 0;
    right: 0;
  }

  @media screen and (max-width: $mq-tablet) {

  }

  @media screen and (max-width: $mq-mobile) {

  }
}
