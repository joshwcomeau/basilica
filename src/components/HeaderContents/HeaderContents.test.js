/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import HeaderContents from './index';

describe('HeaderContents', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HeaderContents />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
