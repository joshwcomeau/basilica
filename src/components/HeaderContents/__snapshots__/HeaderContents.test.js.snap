exports[`HeaderContents renders correctly 1`] = `
<div
  className="header-contents">
  City: Montreal
</div>
`;
